tests/js/server/shell/shell-any-noncluster.js

```diff
@@ -0,0 +1,220 @@
+    /// @brief check entropy of any(), few picks
+    ////////////////////////////////////////////////////////////////////////////////
+
+    testCheckEntropyCollectionFew1: function () {
+      var i, n, l;
+
+      n = 3;
+
+      l = [];
+      for (i = 0; i < n; ++i) {
+        c.save({ value: i });
+        l.push(i);
+      }
+
+      var dist = getDistribution(n * 200, function () {
+        return c.any().value;
+      });
+
+      // make sure we hit everything
+      assertEqual(Object.keys(dist).length, 3);
+    },
+
+    ////////////////////////////////////////////////////////////////////////////////
+    /// @brief check entropy of any(), few picks
+    ////////////////////////////////////////////////////////////////////////////////
+
+    testCheckEntropyCollectionFew2: function () {
+      var i, n, l;
+
+      n = 10;
+
+      l = [];
+      for (i = 0; i < n; ++i) {
+        c.save({ value: i });
+        l.push(i);
+      }
+
+      var dist = getDistribution(n * 100, function () {
+        return c.any().value;
+      });
+
+      // make sure we hit everything
+      assertEqual(Object.keys(dist).length, 10);
+    },
+
+    ////////////////////////////////////////////////////////////////////////////////
+    /// @brief check entropy of any(), more picks
+    ////////////////////////////////////////////////////////////////////////////////
+
+    testCheckEntropyCollectionMore: function () {
+      var i, n, l;
+
+      n = 500;
+
+      l = [];
+      for (i = 0; i < n; ++i) {
+        c.save({ value: i });
+        l.push(i);
+      }
+
+      var dist = getDistribution(n * 100, function () {
+        return c.any().value;
+      });
+
+
+      // make sure we hit enough
+      assertTrue(Object.keys(dist).length >= 20);
+    },
+
+    ////////////////////////////////////////////////////////////////////////////////
+    /// @brief check entropy of any(), with many documents deleted
+    ////////////////////////////////////////////////////////////////////////////////
+
+    testCheckEntropyCollectionHalf: function () {
+      var i, n, l;
+
+      n = 500;
+
+      for (i = 0; i < n; ++i) {
+        c.save({ value: i });
+      }
+
+      // remove 50 % of entries
+      var d = Math.round(n * 0.5);
+      for (i = 0; i < d; ++i) {
+        c.remove(c.any());
+      }
+
+      l = db._query(`FOR d IN ${cn} RETURN d.value`).toArray();
+
+      var dist = getDistribution(n * 50, function () {
+        return c.any().value;
+      });
+
+      // make sure we hit enough
+      assertTrue(Object.keys(dist).length >= 20);
+    },
+
+    ////////////////////////////////////////////////////////////////////////////////
+    /// @brief check entropy of any(), with most documents deleted
+    ////////////////////////////////////////////////////////////////////////////////
+
+    testCheckEntropyCollectionSparse: function () {
+      var i, n, l;
+
+      n = 500;
+
+      for (i = 0; i < n; ++i) {
```

Comment:
please use bulk requests here. Collect documents in an array which you save with one insert after that loop.